### PR TITLE
feat(imgs): enhanced image optimization

### LINKED ISSUE #468: GIFs with image recompression
as we started to recompress all images in articles it seems that for `webp` the recompression of animated GIFs works fine, while for browsers that don't support `webp` they will just get a static first frame as a `pjpeg`.
we should probably in the short-term exclude `.gif`s from the image optimization, and midterm use `mp4` across the board... eg. https://blog.adobe.com/hlx_d2d49af9ed80dad7b83f86719dc7acb782875c19.gif?format=mp4

### PR DESCRIPTION
**Pull request by rofe**
Fix #468 

- exclude gifs from compression
- optimize image sizes for
  - [cards](https://img-opt--theblog--adobe.hlx.page/)
  - [post](https://img-opt--theblog--adobe.hlx.page/en/publish/2020/10/20/photoshop-the-worlds-most-advanced-ai-application-for-creatives.html): hero and body images
  - [homepage](https://img-opt--theblog--adobe.hlx.page/): news and featured article
  - [topic](https://img-opt--theblog--adobe.hlx.page/en/topics/news.html): banner
  - [author](https://img-opt--theblog--adobe.hlx.page/en/authors/adobe-communications-team.html): avatars and pictures

### CODE REVIEW

File: scripts.js [hunk 1]
```diff
@@ -225,6 +218,45 @@ window.adobeid = {
...
+const observer = new MutationObserver(mutations => {
+  mutations.forEach(mutation => {
+    mutation.addedNodes.forEach(node => {
+      // only handle images with src=/hlx_*
+      if (node.tagName === 'IMG' && /\/hlx_/.test(node.src)) {
+        const img = node;
+        let width;
+        if (window.blog.pageType === window.blog.TYPE.TOPIC) {
+          // full width topic banner
+          width = window.innerWidth;
```
Comment (rofe) [inline scripts.js:232]: We should probably lock topic banners at certain fixed sizes based on CSS breakpoints...

**Review by kptdobe — APPROVED** [review]
I know we have no test but hard to tell if the code logic makes sense. "Looks ok".

**Comment by rofe** [comment]
I realize that such a change will require a lot of manual testing, and I appreciate your help in doing so. I added links to relevant test pages to the description.

File: scripts.js (hunk 1 above)
Comment (davidnuescheler) [reply scripts.js:232]: i think anything that generates a broad number of image URLs should really be reconsidered, from a distributed caching standpoint i think this yields very undesirable results... i think we should probably lock in a few image sizes per image type and go with that.

**Comment by rofe** [comment]
I've now changed the impl to use fixed image sizes based on our CSS breakpoints:
- [post](https://img-opt--theblog--adobe.hlx.page/en/publish/2020/10/20/photoshop-the-worlds-most-advanced-ai-application-for-creatives.html):
  - up to 600px window width: all post images 600px
  - more than 600px window width: hero image 1000px, body images 800px
- [topic](https://img-opt--theblog--adobe.hlx.page/en/topics/news.html):
  - up to 600px window width: header banner 600px
  - up to 1200px window width: header banner 1200px
  - more than 1200px window width: header banner 2000px (max)
- [author](https://img-opt--theblog--adobe.hlx.page/en/authors/adobe-communications-team.html):
  - up to 1200px window width: bio pic 124px
  - more than 1200px window width: bio pic 224px
- cards:
  - regular: 512px height, 3:2 crop
  - featured (homepage): 640px height, no crop
  - news box (homepage): 260px height, 3:2 crop

And of course all widths will be multiplied by `window.devicePixelRatio`.

**Review by kptdobe — APPROVED** [review]

**Review by davidnuescheler — APPROVED** [review]

**Pull request merged**